src/Bridge/Doctrine/MongoDbOdm/PropertyInfo/DoctrineExtractor.php

```diff
@@ -65,6 +65,10 @@ public function getTypes($class, $property, array $context = [])
         if ($metadata->hasAssociation($property)) {
             $class = $metadata->getAssociationTargetClass($property);
 
+            if (!\is_string($class) && null === $class) {
```

Comment:
This is some ugly attempt to please PHPStan and I don't know if this will work.
I'm not familiar with PHPStan so if someone knows how to fix it I'm all ears.

This problem here is the interface `$metadata->getAssociationTargetClass` returns a `string` in phpDoc, but implementation of it returns `?string`.

See https://github.com/doctrine/mongodb-odm/blob/ccf26d9e023387e9aae86951186c043b3222fe70/lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php#L1797
and https://github.com/doctrine/persistence/blob/79e97bcb1dfe3b242961253d81b8357a04e96ca1/lib/Doctrine/Persistence/Mapping/ClassMetadata.php#L128

Reply:
The best way to solve this is to add a PHPDoc on the returned variable to force PHPStan to consider it as `?string`.
You can also launch PHPStan locally (`vendor/bin/phpstan analyze .`) in order to verify if it works.

Reply:
Great! It works.
Thanks.